server/item/armour_trim.go

```diff
@@ -0,0 +1,48 @@
+package item
+
+import "github.com/df-mc/dragonfly/server/world"
+
+type ArmourTrim struct {
+	Template ArmourTrimTemplate
+	Material TrimMaterial
+}
+
+type TrimMaterial interface {
```

Comment:
I'd probably suggest renaming this to ArmourTrimMaterial, it's already done for templates and trims can only be applied to armour anyways